mkosi/__init__.py

```diff
@@ -1511,6 +1499,55 @@ def build_uki(
             ]  # fmt: skip
             options += ["--ro-bind", context.workspace / "pesign", context.workspace / "pesign"]
 
+    run(
+        cmd,
+        sandbox=context.sandbox(
+            binary=ukify,
+            options=options,
+            devices=context.config.secure_boot_key_source.type != KeySourceType.file,
+        ),
+    )
+
+
+def build_uki(
+    context: Context,
+    stub: Path,
+    kver: str,
+    kimg: Path,
+    microcodes: list[Path],
+    initrds: list[Path],
+    cmdline: Sequence[str],
+    output: Path,
+) -> None:
+    # Older versions of systemd-stub expect the cmdline section to be null terminated. We can't
+    # embed NUL terminators in argv so let's communicate the cmdline via a file instead.
+    (context.workspace / "cmdline").write_text(f"{' '.join(cmdline).strip()}\x00")
+
+    if not (ukify := context.config.find_binary("ukify", "/usr/lib/systemd/ukify")):
+        die("Could not find ukify")
+
+    cmd: list[PathString] = [
+        "build",
```

Comment:
Maybe move this into `run_ukify()` as well, I doubt we'll ever do something else than `build` and if needed we can just make it configurable